builders/python-builder.psm1

```diff
@@ -41,10 +41,13 @@ class PythonBuilder {
     PythonBuilder ([semver] $version, [string] $architecture, [string] $platform) {
         $this.InstallationTemplatesLocation = Join-Path -Path $PSScriptRoot -ChildPath "../installers"
 
-        $this.HostedToolcacheLocation = $env:AGENT_TOOLSDIRECTORY
-        $this.TempFolderLocation = $env:BUILD_SOURCESDIRECTORY
-        $this.WorkFolderLocation = $env:BUILD_BINARIESDIRECTORY
-        $this.ArtifactFolderLocation = $env:BUILD_STAGINGDIRECTORY
+        New-Item -Force -Type Directory (Join-Path $env:RUNNER_TEMP "artifact")
```

Comment:
Create an additional variable for (Join-Path $env:RUNNER_TEMP "artifact") or move below $this.ArtifactFolderLocation and pass those variables.